opus/application/static_media/js/browse.js

```diff
@@ -424,6 +462,59 @@ var o_browse = {
         });
     }, // end browse behaviors
 
+    // update order arrows right away when user clicks on sorting arrows in pill or table header
+    // sync up arrows in both sorting pill and table header
+    updateOrderIndicator: function(headerOrderIndicator, pillOrderIndicator, isDescending, slug) {
+        let headerOrder = isDescending ? "sort-desc" : "sort-asc";
+        let headerOrderArrow = isDescending ? tableSortUpArrow : tableSortDownArrow;
+        let isPillOrderDesc = isDescending ? "true" : "false";
+        let pillOrderTooltip = isDescending ? "Change to ascending sort" : "Change to descending sort";
+        let pillOrderArrow = isDescending ? pillSortUpArrow : pillSortDownArrow;
+
+        // If header already exists, we update the header arrow, else we do nothing
+        if (headerOrderIndicator && headerOrderIndicator.length !== 0) {
+            headerOrderIndicator.data("sort", `${headerOrder}`);
+            headerOrderIndicator.attr("class", `column_ordering ${headerOrderArrow}`);
+
+            // Reset arrows on rest of table headers
+            // let headers = $(`.dataTable th a:not([data-slug="opusid"], [data-slug=${slug}])`).find("span:last");
+            let headers = $(`.dataTable th a:not([data-slug=${slug}])`).find("span:last");
+            headers.data("sort", "none");
+            headers.attr("class", defaultTableSortArrow);
+        }
+
+        // If pill already exists, we update the pill, else we re-render the pill
+        if (pillOrderIndicator.length !== 0 && slug !== "opusid") {
+            pillOrderIndicator.parent().attr("data-descending", `${isPillOrderDesc}`);
+            pillOrderIndicator.attr("title", `${pillOrderTooltip}`);
+            pillOrderIndicator.find("i").attr("class", `${pillOrderArrow}`);
+        } else {
+            // re-render each pill
+            let listHtml = "";
+            $.each(opus.prefs.order, function(index, orderEntry) {
+                isPillOrderDesc = orderEntry[0] === "-" ? "true" : "false";
+                pillOrderArrow = orderEntry[0] === "-" ? pillSortUpArrow : pillSortDownArrow;
+                orderEntry = orderEntry[0] === "-" ? orderEntry.slice(1) : orderEntry;
+
+                let label = $(`#browse .dataTable th a[data-slug="${orderEntry}"]`).find("span:first").text() || $(`#browse .sort-contents span[data-slug="${orderEntry}"] .flip-sort`).text();
+                let indexOfUnit = label.indexOf("(")
```

Comment:
It seems like you are going to a big effort to get a piece of information that you could just put in the data- bit of the element in the way you want to see it... example:
$(`#browse .dataTable th a[data-slug="${orderEntry}"]`).data("label").  Just a thought, because then you won't have to mess around removing units and all that biz here, it's already there and on the element easy to access.